src/services/services.ts

```diff
@@ -3345,6 +3346,12 @@ export function createLanguageService(
         );
     }
 
+    function getImports(fileName: string): readonly string[] {
+        synchronizeHostData();
+        const file = getValidSourceFile(fileName)
+        return mapDefined(file.imports, i => program.getResolvedModuleFromModuleSpecifier(i, file)?.resolvedModule?.resolvedFileName);

```

Comment:
This will get

- imports written by the user (normal, async, type-only)
- re-exports (`export * from "./foo"`) written by the user
- the implicit tslib and/or JSX factory import added during emit
- import types (`typeof import("./foo")`)
- requires in JS

and it is possible it will contain duplicates. In other words, this is collecting the maximal amount of stuff you could mean by "imports in a file." Not sure what you intend, but I'm guessing you may at least want to deduplicate, and maybe consider dropping the tslib/JSX one?